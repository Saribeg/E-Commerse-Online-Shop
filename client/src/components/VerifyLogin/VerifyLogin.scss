.verify {
  &-block {
    display: flex;
    flex-direction: column;
    justify-content: center;
    align-items: center;
    width: 500px;
    margin: 30px auto;
    border: 2px solid #504e4e;
    border-radius: 5px;
  }

  &-title {
    font-size: 20px;
    font-weight: 600;
    margin-top: 15px;
    margin-bottom: 15px;
  }

  &-text {

  }

  &-link {
    text-decoration: none;
    font-size: 20px;
    font-weight: 600;
    margin-top: 15px;
    margin-bottom: 15px;
  }

}