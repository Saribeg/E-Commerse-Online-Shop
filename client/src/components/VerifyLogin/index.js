import React, {Component} from 'react';
import axios from 'axios';

import {NavLink} from "react-router-dom";

import "./VerifyLogin.scss"

class VerifyLogin extends Component {

    state = {
        success: false,
    }

    componentDidMount() {

        axios.post("/users/verify", {id: this.props.match.params.id})
            .then(res => res.data)
            .then(data => {
                if (data.success) {
                    this.setState({
                        success: true
                    })
                }
            })
            .catch()

    }

    render() {

        return (
            <>
                {
                    (!this.state.success) ?
                        (<div className="verify-block">

                            <p className="verify-title">
                                Unsuccessfull verifying of email
                            </p>

                            <p className="verify-text">
                                You try to verify incorrect email
                            </p>
                            <p className="verify-text">
                                Please open our mail and click on link
                            </p>

                            <NavLink className="verify-link" to="/">
                                Go to main page
                            </NavLink>
                        </div>) :
                        (<div className="verify-block">

                            <p className="verify-title">
                                Successfull verifying of email
                            </p>

                            <p className="verify-text">
                                You account was successfully verified.
                            </p>
                            <p className="verify-text">
                                You can go to main page and log in
                            </p>

                            <NavLink className="verify-link" to="/">
                                Go to main page
                            </NavLink>
                        </div>)
                }
            </>
        )
    }

}

export default VerifyLogin