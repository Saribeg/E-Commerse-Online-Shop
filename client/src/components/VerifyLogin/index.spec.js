import React from 'react';
import VerifyLogin from './index';

const match = {  
        params: {
            id: 1
        }
}

describe('VerifyLogin component', () => {
    it('VerifyLogin component render', () => {
        const wrapper  = shallow(<VerifyLogin match={match}/>)
        expect(wrapper).toMatchSnapshot()
    })
})