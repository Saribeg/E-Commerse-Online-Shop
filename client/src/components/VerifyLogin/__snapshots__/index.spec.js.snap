// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VerifyLogin component VerifyLogin component render 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <VerifyLogin
    match={
      Object {
        "params": Object {
          "id": 1,
        },
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": <div
        className="verify-block"
      >
        <p
          className="verify-title"
        >
          Unsuccessfull verifying of email
        </p>
        <p
          className="verify-text"
        >
          You try to verify incorrect email
        </p>
        <p
          className="verify-text"
        >
          Please open our mail and click on link
        </p>
        <NavLink
          activeClassName="active"
          aria-current="page"
          className="verify-link"
          to="/"
        >
          Go to main page
        </NavLink>
      </div>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <p
            className="verify-title"
          >
            Unsuccessfull verifying of email
          </p>,
          <p
            className="verify-text"
          >
            You try to verify incorrect email
          </p>,
          <p
            className="verify-text"
          >
            Please open our mail and click on link
          </p>,
          <NavLink
            activeClassName="active"
            aria-current="page"
            className="verify-link"
            to="/"
          >
            Go to main page
          </NavLink>,
        ],
        "className": "verify-block",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Unsuccessfull verifying of email",
            "className": "verify-title",
          },
          "ref": null,
          "rendered": "Unsuccessfull verifying of email",
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "You try to verify incorrect email",
            "className": "verify-text",
          },
          "ref": null,
          "rendered": "You try to verify incorrect email",
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Please open our mail and click on link",
            "className": "verify-text",
          },
          "ref": null,
          "rendered": "Please open our mail and click on link",
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "activeClassName": "active",
            "aria-current": "page",
            "children": "Go to main page",
            "className": "verify-link",
            "to": "/",
          },
          "ref": null,
          "rendered": "Go to main page",
          "type": [Function],
        },
      ],
      "type": "div",
    },
    "type": Symbol(react.fragment),
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": <div
          className="verify-block"
        >
          <p
            className="verify-title"
          >
            Unsuccessfull verifying of email
          </p>
          <p
            className="verify-text"
          >
            You try to verify incorrect email
          </p>
          <p
            className="verify-text"
          >
            Please open our mail and click on link
          </p>
          <NavLink
            activeClassName="active"
            aria-current="page"
            className="verify-link"
            to="/"
          >
            Go to main page
          </NavLink>
        </div>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <p
              className="verify-title"
            >
              Unsuccessfull verifying of email
            </p>,
            <p
              className="verify-text"
            >
              You try to verify incorrect email
            </p>,
            <p
              className="verify-text"
            >
              Please open our mail and click on link
            </p>,
            <NavLink
              activeClassName="active"
              aria-current="page"
              className="verify-link"
              to="/"
            >
              Go to main page
            </NavLink>,
          ],
          "className": "verify-block",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Unsuccessfull verifying of email",
              "className": "verify-title",
            },
            "ref": null,
            "rendered": "Unsuccessfull verifying of email",
            "type": "p",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "You try to verify incorrect email",
              "className": "verify-text",
            },
            "ref": null,
            "rendered": "You try to verify incorrect email",
            "type": "p",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Please open our mail and click on link",
              "className": "verify-text",
            },
            "ref": null,
            "rendered": "Please open our mail and click on link",
            "type": "p",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "activeClassName": "active",
              "aria-current": "page",
              "children": "Go to main page",
              "className": "verify-link",
              "to": "/",
            },
            "ref": null,
            "rendered": "Go to main page",
            "type": [Function],
          },
        ],
        "type": "div",
      },
      "type": Symbol(react.fragment),
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;
